src/utilcode/configuration.cpp

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// --------------------------------------------------------------------------------------------------
+// configuration.cpp
+// 
+//
+// Unified method of accessing configuration values from environment variables,
+// registry and config file. See file:../inc/CLRConfigValues.h for details on how to add config values.
+// 
+// --------------------------------------------------------------------------------------------------
+
+#include "stdafx.h"
+
+#include "clrconfig.h"
+#include "configuration.h"
+
+// Information about a configuration knob.
+struct ConfigurationKnob
+{
+    // String representing the knob (e.g. "System.GC.EnableFooGc")
+    LPCWSTR knobName;
+
+    // Value as string (e.g. "true", "1", "Foo")
+    LPCWSTR knobValue;
+};
+
+ConfigurationKnob *knobs = nullptr;
+int numberOfConfigsSet = 0;
+
+void Configuration::InitializeConfigurationKnobs(int numberOfConfigs, LPCWSTR *names, LPCWSTR *values)
+{
+    knobs = new ConfigurationKnob[numberOfConfigs];
+    numberOfConfigsSet = 0;
+
+    for (int i = 0; i < numberOfConfigs; ++i)
+    {
+        _ASSERT(names[i] != nullptr && values[i] != nullptr);
+        if (names[i] != nullptr && values[i] != nullptr)
+        {
+            size_t bufferLength = wcslen(names[i]) + 1;
+            LPWSTR nameBuffer = new WCHAR[bufferLength];
```

Comment:
The caller just created a copy. You should not need to create another one here - just take ownership of the copy created by the caller.